test/fixed_set_test.cpp

```diff
@@ -173,6 +173,15 @@ TEST(FixedSet, EmptySizeFull)
     static_assert(!s4.full());
 }
 
+TEST(FixedSet, MaxSizeDeduction)
+{
+    constexpr auto s1 = make_fixed_set<int>({30, 31});
```

Comment:
`<int>` can be auto-deduced, which is an advantage. Should the examples demonstrate that by skipping the explicit template parameter?

Note:
for map specifically, one would need to specify the first element only:
```C++
    constexpr auto s1 = make_fixed_map({std::pair<int, int>{30, 30}, {31, 54}});
```

(which is something that is needed is `std::map` template deduction as well [link](https://en.cppreference.com/w/cpp/container/map/deduction_guides))

Given this, could we use deduction guides instead of free functions after all?